### ISSUE TITLE
Possible error in logical statement of matrix_dgemm()

### ISSUE THREAD

**Issue by bska** [Ensembles/ert#994]
Function `matrix_dgemm()` ([`devel/libert_util/src/matrix_blas.c`](https://github.com/Ensembles/ert/blob/master/devel/libert_util/src/matrix_blas.c#L110)) contains the following statement on line 184:
```C
if (!ldc >= util_int_max(1 , m)) {
```
This applies the logical NOT operation to `ldc` and then compares the result (0 or 1) to `util_int_max(1,m)`.  Is that really the intended meaning here or did you mean something along the lines of
```C
if (! (ldc >= util_int_max(1, m)))  {
```
or
```C
if (ldc < util_int_max(1, m)) {
```
?

**Comment by joakim-hove** [comment]
Ahhh - the joys of untested; it is correct until .... You are obviously right; I will push a fix. Thank you!

**Comment by bska** [comment]
> I will push a fix

Did this fall by the wayside?  The current sources use the same condition for deciding whether or not to abort the matrix multiply.

**Comment by joakim-hove** [comment]
> Did this fall by the wayside? 

Ehhh - yes it did; completely. But now there is a fix: #1286

**Comment by bska** [comment]
Closing as fixed in #1286.
